op-node/node/node.go

```diff
@@ -138,6 +138,12 @@ func (n *OpNode) initL1(ctx context.Context, cfg *Config) error {
 		}
 		return eth.WatchHeadChanges(n.resourcesCtx, n.l1Source, n.OnNewL1Head)
 	})
+
+	if fallbackClient, ok := l1Node.(*sources.FallbackClient); ok {
+		fallbackClient.RegisterSubscribeFunc(func() (event.Subscription, error) {
+			return eth.WatchHeadChanges(n.resourcesCtx, fallbackClient, n.OnNewL1Head)
```

Comment:
At first, I thought that n.l1Source might have nested calls, because it still calls fallbackClient inside. But I suddenly found that it seems to work, because I have modified the switch code to a separate goroutine. It has been modified